### PR TITLE
eks-prow-build-cluster: secondary node-group

### PR DESCRIPTION
**Pull request by pkprzekwas**
To ease future node upgrades and test Intel based instances, this PR introduces additional node group called `build-green`. Once all resources migrate on this new node group, old node groups will be replaced with a new one called `build-blue`.

In addition, this PR extends IAM policies to enable node group updates. 

/hold
Let's wait till the change is rolled out on prod cluster

/assign @xmudrii 
/cc @ameukam

### CODE REVIEW

File: infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars [hunk 1]
```diff
@@ -35,10 +35,14 @@ cluster_autoscaler_version = "v1.25.0"
...
 node_instance_types = ["r5d.xlarge"]
+
+node_ami_green            = "ami-07e8e7dddc8b3bad9"
+node_instance_types_green = ["r5d.xlarge"]
+
 node_volume_size    = 100
 
 node_min_size                   = 1
-node_max_size                   = 3
+node_max_size                   = 4
```
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars:45]: That was me temporary fiddling with instances while testing AlertManager, we can revert this to 3.

File: infra/aws/terraform/prow-build-cluster/modules/cluster-autoscaler/cluster-autoscaler.tf [hunk 2]
```diff
@@ -345,7 +345,7 @@ resource "kubernetes_manifest" "deployment_kube_system_cluster_autoscaler" {
       "namespace" = "kube-system"
     }
     "spec" = {
-      "replicas" = 1
+      "replicas" = 0
```
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/modules/cluster-autoscaler/cluster-autoscaler.tf:348]: Let's revert this to 1, cluster-autoscaler shouldn't delete instances detached from ASG.

File: infra/aws/terraform/prow-build-cluster/eks.tf [hunk 3]
```diff
@@ -161,6 +161,64 @@ module "eks" {
         update = "180m"
       }
 
+      tags = local.node_group_tags
+    }
+    # Build cluster node group.
+    build-green = {
+      name            = "build-managed-green"
+      description     = "EKS managed node group green used for build nodes"
```
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/eks.tf:169]:
```suggestion
      description     = "EKS managed node group called green used to facilitate node rotations/rollouts"
```

**Review by xmudrii — APPROVED** [review]
Some nits, in general LGTM.
/lgtm
/approve

**Comment by xmudrii** [comment]
@pkprzekwas What's the current status of this rollout?

**Comment by pkprzekwas** [comment]
> @pkprzekwas What's the current status of this rollout?

I was waiting for some approval. I can start the process after addressing your comments.

**Comment by pkprzekwas** [comment]
Changes have been rolled out on Prod and Canary. Looking for another lgtm to merge this.

File: infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars [hunk 4]
```diff
@@ -33,13 +33,22 @@ cluster_version            = "1.25"
...
 
 # Ubuntu EKS optimized AMI: https://cloud-images.ubuntu.com/aws-eks/
-node_ami            = "ami-07e8e7dddc8b3bad9"
-node_instance_types = ["r5d.xlarge"]
-node_volume_size    = 100
+node_ami_blue            = "ami-07e8e7dddc8b3bad9"
+node_instance_types_blue = ["r5d.xlarge"]
+
+node_min_size_blue     = 0
+node_max_size_blue     = 3
```
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars:40]: Can we have max size 0?

**Review by xmudrii — COMMENTED** [review]
@pkprzekwas What's the best way that we document why we need blue and green node groups?

**Comment by pkprzekwas** [comment]
> @pkprzekwas What's the best way that we document why we need blue and green node groups?

I was planning to prepare a separate document describing both the motivation and process for that.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**